upgrade/upgrade-scripts/upgrade-container

```diff
@@ -505,6 +505,63 @@ function migrate_configuration() {
 	EOF
 }
 
+function do_upgrade_container_in_place() {
+	run "$IMAGE_PATH/execute" -p "$(get_platform)" ||
+		die "'$IMAGE_PATH/execute' failed in '$CONTAINER'"
+
+	run /bin/systemctl reload delphix-platform ||
+		die "'systemctl reload delphix-platform' failed in '$CONTAINER'"
+
+	run /bin/systemctl restart delphix-platform ||
+		die "'systemctl restart delphix-platform' failed in '$CONTAINER'"
+}
+
+function do_upgrade_container_not_in_place() {
+	run "$IMAGE_PATH/execute" -p "$(get_platform)" ||
+		die "'$IMAGE_PATH/execute' failed in '$CONTAINER'"
+
+	run /bin/systemctl start delphix-platform ||
+		die "'systemctl start delphix-platform' failed in '$CONTAINER'"
+
+	migrate_configuration ||
+		die "failed to migrate configuration for '$CONTAINER'"
+}
+
+function get_type() {
+	local root origin
+
+	root=$(zfs list -Hpo name /var/lib/machines/"${CONTAINER}")
+	[[ -n "${root}" ]] || die "Failed to obtain root filesystem for container '${CONTAINER}'"
+	origin=$(zfs get -Hpo value origin "${root}")
+
+	case "${origin}" in
+	-)
+		echo "not-in-place"
+		;;
+	*)
+		echo "in-place"
+		;;
+	esac
+}
+
+function do_upgrade_container() {
+	local type
+	type=$(get_type)
+	case $type in
```

Comment:
can we do this:
```
case "$(get_type)" in
```
I'm not sure if that's valid bash syntax, but if so, it'll allow us to remove the `type` variable.